Convert text constraint definitions to JSON

ZeLonewolf opened the pull request:
Fixes #118 and moves us towards #146 and #70.

This PR converts shield text constraints to JSON, which allows the shield definitions to be pure JSON.  There is still some work we could do to reduce repetition in the JSON structure, but there's no longer any function pointers in the definition.

I also added npm scripts to generate the shield text definitions out to JSON at the command line.

claysmalley commented:
Did something change with `southHalfEllipseTextConstraint`? Before/after/diff:

![old-h201](https://user-images.githubusercontent.com/1732117/216444314-213ee299-e562-4d04-80af-b1c4a351337f.png) ![new-h201](https://user-images.githubusercontent.com/1732117/216444318-4f81f116-8eee-4da2-b0f3-33e9091cfeb2.png) ![diff-h201](https://user-images.githubusercontent.com/1732117/216444320-eebd7c0b-4094-4a40-879a-ae1b022fdbfb.png)
![old-a26a7](https://user-images.githubusercontent.com/1732117/216444317-b4271c25-c45a-4064-af93-b81edf83d65a.png) ![new-a26a7](https://user-images.githubusercontent.com/1732117/216444319-46b2fbc9-be73-4c91-a1a2-f46f82cd7850.png) ![diff-a26a7](https://user-images.githubusercontent.com/1732117/216444321-117d664b-fcfb-4a2c-b99a-6c4e1212914c.png)

ZeLonewolf commented:
Yep, that looks pretty wrong

ZeLonewolf commented:
So the right hand graphic is a correct south-half ellipse and the original is wrong.  So somehow I fixed the bug...

I think for fishhead we actually want a rounded rect constraint for fishhead, because of its shape, which I changed here:

![image](https://user-images.githubusercontent.com/3254090/216857701-fd476736-b5d3-4bde-bdfb-78dfedb7a4a9.png)
